### PR TITLE
RN-156: Added startDate and endDate selectors to the viz-builder

### PR DESCRIPTION
**Pull request by rohan-bes**
### Issue: RN-156

Needed to do this to test the changes in https://github.com/beyondessential/tupaia/pull/4406

Such a simple feature, and it makes the viz-building experience so much better. So I feel it's worth it

### CODE REVIEW

File: packages/admin-panel/src/VizBuilderApp/components/PreviewOptions.js [hunk 1]
```diff
@@ -151,6 +173,24 @@ export const PreviewOptions = () => {
     setLocation(value.code);
   };
 
+  const handleChangeStartDate = date => {
+    if (!date) {
```
Comment (biaoli0) [inline packages/admin-panel/src/VizBuilderApp/components/PreviewOptions.js:177]: I suggest to set value when it is null. User case is user wants to see all data from all period, they will delete the previous date input.

File: packages/admin-panel/src/VizBuilderApp/components/PreviewOptions.js [hunk 2]
```diff
@@ -151,6 +173,24 @@ export const PreviewOptions = () => {
...
+      return;
+    }
+
+    setSelectedStartDate(date.toISOString());
+    setStartDate(date.toISOString());
+  };
+
+  const handleChangeEndDate = date => {
+    if (!date) {
```
Comment (biaoli0) [inline packages/admin-panel/src/VizBuilderApp/components/PreviewOptions.js:186]: same here

**Review by biaoli0 — CHANGES_REQUESTED** [review]
@rohan-bes Thanks for adding the date selector in viz builder! It will make everyone's life easier!

Just request a minor change to handle a special user case.

File: packages/admin-panel-server/src/routes/FetchReportPreviewDataRoute.ts [hunk 3]
```diff
@@ -29,6 +29,8 @@ export type FetchReportPreviewDataRequest = Request<
   {
     entityCode: string;
     hierarchy: string;
+    startDate: string;
+    endDate: string;
```
[the remark this answers is not in the document]
Comment (rohan-bes) [reply packages/admin-panel-server/src/routes/FetchReportPreviewDataRoute.ts:33]: Ah I see, will make it optional I think 👍

**Review by biaoli0 — APPROVED** [review]
LGTM!

**Pull request merged**